include/trial/protocol/json/partial/copy_and_look_ahead.hpp

```diff
@@ -0,0 +1,176 @@
+#ifndef TRIAL_PROTOCOL_JSON_PARTIAL_COPY_AND_LOOK_AHEAD_HPP
+#define TRIAL_PROTOCOL_JSON_PARTIAL_COPY_AND_LOOK_AHEAD_HPP
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (C) 2021 Vinícius dos Santos Oliveira <vini.ipsmaker@gmail.com>
+//
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <trial/protocol/json/partial/skip.hpp>
+
+namespace trial
+{
+namespace protocol
+{
+namespace json
+{
+namespace partial
+{
+
+namespace detail {
+template<class CharT>
+struct CopyAndLookAheadCollector
```

Comment:
The naming convention is similar to the standard library and Boost, where CamelCase is only used for template parameter types. Normal types are using underscore_case.

Reply:
Fixed: https://github.com/breese/trial.protocol/pull/57/commits/d963d4972d6d3c93f6b697ab8eaa2263a54d4f88